Bunkum.HttpServer/Endpoints/Middlewares/MainMiddleware.cs

```diff
@@ -221,9 +202,21 @@ public void HandleRequest(ListenerContext context, Lazy<IDatabaseContext> databa
                         }
                         else
                         {
-                            // We don't know what this param is or what to do with it, so pass in null.
-                            // Better than not calling the endpoint and throwing an exception.
-                            invokeList.Add(null);
+                            // Ask all services to try to provide an argument for this parameter.
+                            object? arg = null;
+                            
+                            List<Service>.Enumerator services = this._services.GetEnumerator();
+                            while (arg == null)
+                            {
+                                if (!services.MoveNext()) break;
+                                arg = services.Current.AddParameterToEndpoint(context, param, database);
+                            }
+                            
+                            services.Dispose();
+
+                            // Add the arg even if null, as even if we don't know what this param is or what to do with it,
+                            // it's probably better than not calling the endpoint and throwing an exception.
+                            invokeList.Add(arg);
```

Comment:
there should be a check here that breaks a debugger, if one is attached and its null, to help catch cases of accidental UB, without causing exceptions at runtime (although arguably passing null when the person is expecting non-null could cause harder to debug excpetions too, so maybe we should just throw an exception here if null?)

Reply:
Throwing an exception can cause bad behavior since this can be necessary sometimes, see https://github.com/LittleBigRefresh/Refresh/pull/13#discussion_r1086131790

I think its more appropriate to log a warning here.